src/access/acp.test.ts

```diff
@@ -3961,3 +4064,2232 @@ describe("getAgentAccessAll", () => {
+        "https://some.pod/resource",
+        "https://some.pod/resource?ext=acr",
+        {
+          policies: {
+            "https://some.pod/resource?ext=acr#policy": {
+              allow: { read: true, append: true, write: true },
+              allOf: {
+                "https://some.pod/resource?ext=acr#rule": {
+                  [acp.agent]: [webId],
+                },
+              },
+            },
+          },
+          memberPolicies: {},
+          acrPolicies: {
+            "https://some.pod/resource?ext=acr#acrPolicy": {
+              allow: { read: true, write: true },
+              allOf: {
+                "https://some.pod/resource?ext=acr#acrRule": {
+                  [acp.agent]: [webId],
+                },
+              },
+            },
+          },
+          memberAcrPolicies: {},
+        }
+      );
+
+      const updatedResourceWithAcr = internal_setActorAccess(
+        resourceWithAcr,
+        acp.agent,
+        webId,
+        {
+          controlRead: false,
+        }
+      );
+
+      expect(
+        internal_getActorAccess(resourceWithAcr, acp.agent, webId)
+      ).toStrictEqual({
+        read: true,
+        append: true,
+        write: true,
+        controlRead: true,
+        controlWrite: true,
+      });
+      expect(
+        internal_getActorAccess(updatedResourceWithAcr!, acp.agent, webId)
+      ).toStrictEqual({
+        read: true,
+        append: true,
+        write: true,
+        controlRead: false,
+        controlWrite: true,
+      });
+    });
+
+    it("preserves conflicting Control access defined for a different actor that is defined with the same Rule as applies to the given actor", () => {
+      const resourceWithAcr = mockResourceWithAcr(
+        "https://some.pod/resource",
+        "https://some.pod/resource?ext=acr",
+        {
+          policies: {},
+          memberPolicies: {},
+          acrPolicies: {
+            "https://some.pod/resource?ext=acr#policy": {
+              allow: { read: true },
+              allOf: {
+                "https://some.pod/resource?ext=acr#rule": {
+                  [acp.agent]: [
+                    webId,
+                    "https://some-other.pod/other-profile#me",
+                  ],
+                },
+              },
+            },
+          },
+          memberAcrPolicies: {},
+        }
+      );
+
+      const updatedResourceWithAcr = internal_setActorAccess(
+        resourceWithAcr,
+        acp.agent,
+        webId,
+        {
+          controlRead: false,
+        }
+      );
+
+      expect(
+        internal_getActorAccess(
+          resourceWithAcr,
+          acp.agent,
+          "https://some-other.pod/other-profile#me"
+        )
+      ).toStrictEqual({
+        controlRead: true,
+      });
+      expect(
+        internal_getActorAccess(updatedResourceWithAcr!, acp.agent, webId)
+      ).toStrictEqual({
+        controlRead: false,
+      });
```

Comment:
Same remark as for setting access (I'll not repeat it everywhere, because I guess the same explanation will apply :) )